source/MaterialXTest/ShaderValid.cpp

```diff
@@ -367,192 +365,68 @@ static void runValidation(const std::string& shaderName, std::vector<mx::Element
-    swizzle1->setConnectedNode("in", texcoord1);
-    swizzle1->setParameterValue("channels", std::string("xy0"));
-    mx::OutputPtr texcoord1_output = nodeGraph->addOutput("texcoord1_output", "vector3");
-    texcoord1_output->setConnectedNode(swizzle1);
-    outputList.push_back(texcoord1_output);
-
-    // Second UV stream test
-    mx::NodePtr texcoord2 = nodeGraph->addNode("texcoord", "texcoord2", "vector2");
-    texcoord2->setParameterValue("index", 1, "integer");
-    mx::NodePtr swizzle2 = nodeGraph->addNode("swizzle", "uv_set1", "vector3");
-    swizzle2->setConnectedNode("in", texcoord2);
-    swizzle2->setParameterValue("channels", std::string("xy0"));
-    mx::OutputPtr texcoord2_output = nodeGraph->addOutput("texcoord2_output", "vector3");
-    texcoord2_output->setConnectedNode(swizzle2);
-    outputList.push_back(texcoord2_output);
-
-    // Image test
-    mx::FilePath imagePath = mx::FilePath::getCurrentPath() / mx::FilePath("documents/Images/MaterialXLogo.exr");
-    std::string imageName = imagePath.asString();
-    mx::NodePtr image1 = nodeGraph->addNode("image", "image1", "color3");
-    image1->setParameterValue("file", imageName, "filename");
-    mx::OutputPtr image_output = nodeGraph->addOutput("image1_output", "vector3");
-    image_output->setConnectedNode(image1);
-    outputList.push_back(image_output);
-
-    // Create a validator
-    bool orthographicView = true;
-    mx::GlslValidatorPtr validator = createValidator(orthographicView, "shaderball.obj", log);
-
-    // Set up shader generator
-    mx::ShaderGeneratorPtr shaderGenerator = mx::GlslShaderGenerator::create();
-    shaderGenerator->registerSourceCodeSearchPath(searchPath);
-
-    // Run validation
-    runValidation(nodeGraph->getName(), outputList, validator, shaderGenerator, orthographicView, doc, log);
+    auto it = toMatch.begin();
+	  return str.size() >= toMatch.size() && std::all_of(std::next(str.begin(),str.size() - toMatch.size()), str.end(),
+        [&it](const char & c)
+        {
+            return ::tolower(c) == ::tolower(*(it++))  ;
+        });
 }
 
-TEST_CASE("GLSL shading", "[shadervalid]")
+TEST_CASE("GLSL MaterialX documents", "[shadervalid]")
 {
-#ifdef LOG_TO_FILE
-    std::ofstream logfile("log_shadervalid_glsl_shading.txt");
-    std::ostream& log(logfile);
-#else
-    std::ostream& log(std::cout);
-#endif
-    mx::DocumentPtr doc = mx::createDocument();
-
-    mx::FilePath searchPath = mx::FilePath::getCurrentPath() / mx::FilePath("documents/Libraries");
-    loadLibraries({ "stdlib", "sxpbrlib" }, searchPath, doc);
-
-    // Create a validator
-    bool orthographicView = true;
-    mx::GlslValidatorPtr validator = createValidator(orthographicView, "shaderball.obj", log);
-
-    // Create shader generator
-    mx::ShaderGeneratorPtr shaderGenerator = mx::GlslShaderGenerator::create();
-    shaderGenerator->registerSourceCodeSearchPath(searchPath);
-
-    // Set up lighting
```

Comment:
Might want to leave this here for now. Light is added in explicitly and if this is data driven then we need a way to determine when lighting is required. Maybe if the element type is a shader... tbd.